packages/@aws-cdk/aws-ecs/lib/cluster.ts

```diff
@@ -216,16 +210,16 @@ export class Cluster extends Resource implements ICluster {
     // since it's harmless, but we'd prefer not to add unexpected new
     // resources to the stack which could surprise users working with
     // brown-field CDK apps and stacks.
-    Aspects.of(this).add(new MaybeCreateCapacityProviderAssociations(this, id, this._asgCapacityProviders));
+    Aspects.of(this).add(new MaybeCreateCapacityProviderAssociations(this, id, this._capacityProviderNames));
```

Comment:
Passing `id` onward here is a bit dangerous. Ideally you'd want to give this its own identifier.

Although that might be impossible to change now for backwards compatibility reasons, as changing the `id` now will cause resources to be replaced for people who already deployed the previous version.

Reply:
Yes, I had thought about changing this to something more unique to the CPAssociation so it's less confusing but thought that would be breaking, and also was running into the strange issue of this singleton class being created twice for some reason. I guess we'll leave it as is?